### PR TITLE
[FEATURE] Add min and max of timestamp cols to Column Descriptive Metrics

### PR DESCRIPTION
**Pull request by rreinoldsc**
Add computation of min/max metrics for timestamp columns for Column Descriptive Metrics. Mean and Median were also considered but these are not supported out of the box by snowflake so implementation is deferred.

See also https://github.com/great-expectations/great_expectations/pull/9158 for additional test refactoring.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py [hunk 1]
```diff
@@ -212,6 +219,49 @@ def _get_numeric_column_metrics(
 
         return metrics
 
+    def _get_timestamp_column_metrics(
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py]: this should be refactored

File: great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py [hunk 2]
```diff
@@ -280,6 +330,27 @@ def _get_numeric_column_names(
...
 
+    def _get_timestamp_column_names(
+        self,
+        batch_request: BatchRequest,
+        exclude_column_names: List[str],
+    ) -> list[str]:
+        """Get the names of all numeric columns in the batch."""
+        validator = self.get_validator(batch_request=batch_request)
+        domain_builder = ColumnDomainBuilder(
+            include_semantic_types=[SemanticDomainTypes.DATETIME],
```
Comment (rreinoldsc) [inline great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py]: DATETIME is all I changed, should be refactored

File: tests/experimental/metric_repository/test_column_descriptive_metrics_metric_retriever.py [hunk 3]
```diff
@@ -58,8 +58,11 @@ def test_get_metrics():
     mock_batch_request = Mock(spec=BatchRequest)
 
     with mock.patch(
-        f"{ColumnDomainBuilder.__module__}.{ColumnDomainBuilder.__name__}.get_effective_column_names",
+        f"{ColumnDescriptiveMetricsMetricRetriever.__module__}.{ColumnDescriptiveMetricsMetricRetriever.__name__}._get_numeric_column_names",
```
Comment (anthonyburdi) [inline tests/experimental/metric_repository/test_column_descriptive_metrics_metric_retriever.py:61]: Note that there is an upcoming PR to refactor these tests and remove some duplication via fixtures: https://github.com/great-expectations/great_expectations/pull/9158

**Review by JennyTee — APPROVED** [review]
lgtm, though this is the first time I've looked at the Core CDM logic.

Only concern is that it's not easy to see what will need to be refactored when we support more than just Snowflake...but that's not a blocker

File: tests/experimental/metric_repository/test_column_descriptive_metrics_metric_retriever_integration.py [hunk 4]
```diff
@@ -189,6 +189,20 @@ def test_get_metrics(
...
+            exception=None,
+            column="datetime",
+        ),
+        ColumnMetric[str](
+            batch_id=batch_id,
+            metric_name="column.max",
+            value=Timestamp("2020-01-03 00:00:00"),
+            exception=None,
+            column="datetime",
+        ),
```
Comment (anthonyburdi) [inline tests/experimental/metric_repository/test_column_descriptive_metrics_metric_retriever_integration.py:205]: @rreinold Note that timestamp min/max metrics are not always returned as strings, but are for Snowflake (see here Pandas `Timestamp`). We will need to address when more datasources are supported for CDM.

**Pull request merged**